pvl/__init__.py

```diff
@@ -122,16 +123,19 @@ def decode_by_char(f: io.RawIOBase) -> str:
     The *f* stream will have one character or byte at a time read from it,
     and will attempt to decode each to a string and accumulate
     those individual strings together.  Once the end of the file is found
-    or an element can no longer be decoded, the accumulated string will
+    or an element can no longer be decoded as UTF, the accumulated string will
     be returned.
     """
     s = ""
     try:
         for elem in iter(lambda: f.read(1), b""):
             if isinstance(elem, str):
+                if elem == "":
+                    break
                 s += elem
             else:
                 s += elem.decode()
```

Comment:
One thing we could do is pass encoding into the constructor and then use `elem.decode(self.encoding)` here.

Reply:
Yes, absolutely.  Somewhere in the long diatribe of my comments on #93, I suggested that.  We could certainly add an `encoding=` parameter to `pvl.load()`.  It doesn't stop the user from having to "know" that their text file is in some non-ASCII, non-UTF encoding scheme, but if they do, this would be a helper, and easy to implement.  I'll work it.